meta-resin-common/classes/image_types_resin.bbclass

```diff
@@ -264,3 +264,14 @@ IMAGE_CMD_docker () {
     DOCKER_IMAGE=$(${IMAGE_CMD_TAR} -cv -C ${IMAGE_ROOTFS} . | docker import -)
     docker save ${DOCKER_IMAGE} > ${IMGDEPLOYDIR}/${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.docker
 }
+
+IMAGE_TYPEDEP_hostapp-ext4 = "docker"
```

Comment:
I think this should be:

`IMAGE_TYPEDEP_hostapp-ext4 = "docker ext4"`

for older (i.e. Krogoth) Poky versions. This should fix errors like:

NOTE: Preparing RunQueue
ERROR: Task do_bootimg in /yocto/resin-board/build/../layers/meta-resin/meta-resin-common/recipes-core/images/resin-image.bb depends upon non-existent task do_image_ext4 in /yocto/resin-board/build/../layers/meta-resin/meta-resin-common/recipes-core/images/resin-image.bb
ERROR: Command execution failed: 1

Reply:
Which board uses krogoth? This will help me figure out the problem

Reply:
up-board for example.
The fix should be as above

Reply:
I don't think the fix is correct. My image type doesn't depend on the ext4 image type

Reply:
It does not but without it, on pre-morty poky, it will complain with the above error. It's probably linked to changes in image types dependencies which have been changed in poky over the various releases (I haven't looked long enough to pinpoint the exact yocto change that triggers this error on krogoth - and I expect this issue to be present on yocto jethro too)

Reply:
fixed. The problem was wrong inheritance order of resin-image